packages/core/test/insured-bridge/ArbitrumMessenger.js

```diff
@@ -0,0 +1,229 @@
+const hre = require("hardhat");
+const { runDefaultFixture, ZERO_ADDRESS } = require("@uma/common");
+const { getContract } = hre;
+const { utf8ToHex, toWei } = web3.utils;
+
+const { deployContractMock } = require("./helpers/SmockitHelper");
+
+const { assert } = require("chai");
+
+// Tested contracts
+const AVM_InboxMock = getContract("AVM_InboxMock");
+const ArbitrumMessenger = getContract("ArbitrumMessenger");
+const BridgeAdmin = getContract("BridgeAdmin");
+const BridgePool = getContract("BridgePool");
+const Timer = getContract("Timer");
+const Finder = getContract("Finder");
+const BridgeDepositBox = getContract("BridgeDepositBoxMock");
+const IdentifierWhitelist = getContract("IdentifierWhitelist");
+const AddressWhitelist = getContract("AddressWhitelist");
+
+// Contract objects
+let arbitrumMessenger;
+let bridgeAdmin;
+let finder;
+let l1InboxMock;
+let depositBox;
+let identifierWhitelist;
+let collateralWhitelist;
+let timer;
+
+// Test function inputs
+const defaultGasLimit = 1_000_000;
+const defaultGasPrice = toWei("1", "gwei");
+const defaultIdentifier = utf8ToHex("IS_CROSS_CHAIN_RELAY_VALID");
+const defaultLiveness = 7200;
+const defaultProposerBondPct = toWei("0.05");
+const lpFeeRatePerSecond = toWei("0.0000015");
+const defaultBridgingDelay = 60;
+const chainId = "10";
+let l1Token;
+let l2Token;
+let bridgePool;
+
+describe("ArbitrumMessenger integration with BridgeAdmin", () => {
+  let accounts, owner, rando, rando2, depositBoxImpersonator;
+
+  before(async function () {
+    accounts = await web3.eth.getAccounts();
+    [owner, rando, rando2, depositBoxImpersonator] = accounts;
+    l1Token = rando;
+    l2Token = rando2;
+    await runDefaultFixture(hre);
+    timer = await Timer.deployed();
+    finder = await Finder.deployed();
+    identifierWhitelist = await IdentifierWhitelist.deployed();
+    collateralWhitelist = await AddressWhitelist.deployed();
+    await identifierWhitelist.methods.addSupportedIdentifier(defaultIdentifier).send({ from: owner });
+  });
+  beforeEach(async function () {
+    l1InboxMock = await deployContractMock("AVM_InboxMock", {}, AVM_InboxMock);
```

Comment:
perfect use of this. I'd like to refactor the `deployContractMock` in the future to make the params a bit cleaner but for now it's fine